dts/x86/intel/raptor_lake.dtsi

```diff
@@ -496,6 +496,16 @@
 			status = "okay";
 		};
 
+		rtc: rtc@70 {
+			compatible = "motorola,mc146818-rtc";
```

Comment:
@bjarki-trackunit wrote:
> I think this is the most proper change to align the devices and drivers, no motorola,mc146818 and motorola,mc146818-rtc, just the first one, using the better fit RTC driver

"no motorola,mc146818 and motorola,mc146818-rtc, just the first one" - maybe I'm misunderstanding that, but "just the first one" refers to "motorola,mc146818". What exactly is the conflict if two drivers try to use the same compat string? That they'd have conflicting binding files? @bjarki-trackunit could you clarify what you meant?

Reply:
@jhedberg

>  it didn't feel like it's a straight cut answer to say "nobody will ever want to use the counter API if the RTC API is available"

I'm saying that: if the RTC drivers are enabled, which is an explicit `CONFIG_RTC=y`, and the counter drivers with `CONFIG_COUNTER=y`, RTC hardware should use the RTC driver, not the counter driver. Otherwise, only enable the counter drivers, and that will be used instead. I do think that will be the expected and preferred behavior in most if not all cases.

That said, if one really wants to select the counter driver for an RTC, and has enabled the RTC drivers, you can explicitly configure the drivers like `CONFIG_RTC_MC146818=n` `CONFIG_COUNTER_CMOS=y`